### PR TITLE
Clean up dependency lists

### PR DESCRIPTION
**Pull request by vyasr**
cudf is not actually used anywhere in tests, only in a single notebook. Similarly, dask is only used by legate tests

### CODE REVIEW

File: dependencies.yaml
Comment (bdice) [inline dependencies.yaml:16]: Alphabetize.
```suggestion
      - notebooks
      - py_version
      - run
      - test_python
```

File: dependencies.yaml
Comment (bdice) [inline dependencies.yaml:231]: Add the fix for `update-version.sh` from #224.

**Review by bdice — CHANGES_REQUESTED** [review]

File: python/pyproject.toml
Comment (bdice) [inline python/pyproject.toml:44]: It seems like `distributed` isn't used in `kvikio` -- only the `legate_kvikio` benchmarks.

File: dependencies.yaml
Comment (vyasr) [reply dependencies.yaml:231]: There isn't actually an update-version.sh fix in #224 since cudf was added back as a dependency, but it looks like I can at least remove some of it so I'll do that now.

Comment (bdice) [reply dependencies.yaml:231]: My mistake. I misread a0a7b596a99a99c34b85ac69acbaed6999b5ea94, you were adding it _back_ after removing it.

File: dependencies.yaml
Comment (bdice) [inline dependencies.yaml:224]: We need `dask` in `test_python`, just not `distributed`.

See imports of `dask` in `python/benchmarks/single-node-io.py` and `python/tests/test_benchmarks.py`.
```suggestion
          - cuda-python>=11.7.1,<12.0
          - &dask dask >=2022.05.2
```

File: dependencies.yaml
Comment (bdice) [inline dependencies.yaml:231]:
```suggestion
          - *dask
```

**Review by bdice — COMMENTED** [review]
A couple small suggestions.

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
LGTM. I’ll merge to unblock #224.

**Comment by bdice** [comment]
Just kidding. I see we need several reviews first.

**Review by jakirkham — APPROVED** [review]

**Review by wence- — APPROVED** [review]
lgtm

**Review by raydouglass — APPROVED** [review]

**Comment by vyasr** [comment]
We need @jakirkham to re-approve (or @madsbk to approve) on behalf of legate codeowners before this will merge, now.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**
